### DISCUSSION TITLE
Better Documentation

### DISCUSSION THREAD

**Discussion by RajvirSingh1313** [asmyshlyaev177/react-horizontal-scrolling-menu#166] [General]
Hey @asmyshlyaev177 , I liked your project, and currently trying to add it to my project, but the problem is the starter code you provided is a little hard to understand & to integrate into any project quickly. I hope you can guide me to some good resources to understand it better, and I am looking forward to contributing to it. As I really liked it.

Looking forward to your suggestion,
With regards,
Rajvir

**Comment by asmyshlyaev177** [comment]
Hi, you can describe which parts hard to understand and how can improve it.

**Comment by maartenvdvlugt** [comment]
Hi @asmyshlyaev177, what I am missing in the documentation is a way to trigger 'scrollToItem()' from outside of the ScrollMenu component, for instance a function that doesn't have this context. 

I have been trying to accomplish this, but with no luck so far. Guess this should be possible, right?

The case is that a user by pressing a button that is rendered outside of the ScrollMenu component can set the ScrollMenu back to it's initial settings as set by onInit. 

Thanks

**Comment by RajvirSingh1313** [comment]
> Hi, you can describe which parts hard to understand and how can improve it.

Hi @asmyshlyaev177, Thanks for replying,

It is hard to integrate the example into my code like I can't just copy and past it into any part of my code and it will just work 

If you can make a quick start code, with everything set up, the developer will just need to just fill it. It would be easier for developers to get started quickly.

You can improve it by adding a simple one-component example, at the top of the readme, as the current top example is a bit overwhelming for me. I hope you got my point.

**Comment by asmyshlyaev177** [comment]
> > Hi, you can describe which parts hard to understand and how can improve it.
> 
> Hi @asmyshlyaev177, Thanks for replying,
> 
> It is hard to integrate the example into my code like I can't just copy and past it into any part of my code and it will just work
> 
> If you can make a quick start code, with everything set up, the developer will just need to just fill it. It would be easier for developers to get started quickly.
> 
> You can improve it by adding a simple one-component example, at the top of the readme, as the current top example is a bit overwhelming for me. I hope you got my point.

Maybe it's too much boilerplate, i make it like this so it easy to maintain and customize component in many ways.
First version was all-in-one combine, but it's got over complicated and buggy over time.
Not sure if i can make example simpler, but if you help with will be great.

**Comment by asmyshlyaev177** [comment]
> Hi @asmyshlyaev177, what I am missing in the documentation is a way to trigger 'scrollToItem()' from outside of the ScrollMenu component, for instance a function that doesn't have this context.
> 
> I have been trying to accomplish this, but with no luck so far. Guess this should be possible, right?
> 
> The case is that a user by pressing a button that is rendered outside of the ScrollMenu component can set the ScrollMenu back to it's initial settings as set by onInit.
> 
> Thanks

Ok, will think how to implement it, currently can't do it outside of context.

**Comment by RajvirSingh1313** [comment]
@asmyshlyaev177 Sure, I will try to make a simpler example, so anyone would be able to just copy and past it into their example. I know it is a little hard to maintain, But if you can try to make a simple example for just a simple use case, then it would be great. And I will help you after this week. As I am a little busy this week. Anyways have a good day, and I will make sure to make a blog on this tool; So others can know and can contribute to making it better.

**Comment by asmyshlyaev177** [comment]
@maartenvdvlugt I have implemented way to access function outside of context - https://codesandbox.io/s/react-horizontal-scrolling-menu-v2-apiref-vdr0d?file=/src/index.tsx  . I hope it's ok.
